internal/gateway/certificate.go

```diff
@@ -0,0 +1,66 @@
+package gateway
```

Comment:
Maybe we could have a more modular structure for those resources.
I see that it is the same as for Istio resources, but maybe this wasn't exactly the best, as it's harder to see now which GO file corresponds to which resource.
I think that having it as separate directories might be better

Reply:
Do you mean moving all YAMLs to a separate directory or have scoped folders (e.g. for certificate, gateway)? 

I didn't want to move certificates, dns to a separate package, yet. Since they are only used in the scope of the gateway and I didn't want to do too much premature optimisation. I can see, that this might be more helpful once we also support custom gateways defined through Gateway CR.

Reply:
I meant having it under scoped folders